### ISSUE TITLE
Wrong triggering rules declaration-empty-line-before in less

### ISSUE THREAD

**Issue by Tom910** [stylelint/stylelint#1911]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

It a bug, incorrectly triggered a rule in LESS

> Which rule, if any, is this issue related to?

declaration-empty-line-before

> What CSS is needed to reproduce this issue?

LESS

```less
.mixin () {
    content: '';
}

.class {
  .mixin();
  margin: 32px;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
"declaration-empty-line-before": [ "always", {
      except: [
        "after-declaration",
        "first-nested",
      ],
      ignore: [
        "after-comment",
        "inside-single-line-block",
      ],
    } ],
```

> Which version of stylelint are you using?

stylelint@7.2.0
stylelint-config-standard@13.0.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with stylelint test.less --syntax=less

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to LESS mixins.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
1.less
 7:5  ✖  Expected empty line before declaration   declaration-empty-line-before
```

**Comment by jeddy3** [comment]
@Tom910 Thank you for using the issue template.

I think the rule is behaving as expected.

```css
.mixin () {
    content: '';
}

.class {
  .mixin();
  margin: 32px;
}
```

Line *7* is a declaration without an empty line before it. None of the `ignore` or `except` optional secondary options are applicable here i.e. it's not the first nested, after a declaration, after a comment nor inside a single line block. Therefore it should warn about the missing empty line.

You should separate your mixins and declarations with an empty line if using this configuration.

If you don't want an empty line, then try experimenting with the `"never"` primary option.
